pkg/cloudprovider/provider/vsphere/helper.go

```diff
@@ -179,6 +179,75 @@ func createLinkClonedVm(vmName, vmImage, datacenter, clusterName, folder string,
 	_, err = clonedVmTask.WaitForResult(ctx, nil)
 	return err
 }
+
+func updateNetworkForVM(ctx context.Context, vm *object.VirtualMachine, currentNetName string, newNetName string) error {
+	newNet, err := getNetworkFromVM(ctx, vm, newNetName)
+	if err != nil {
+		return err
+	}
+
+	netDev, currentBacking, err := getNetworkDeviceAndBackingFromVM(ctx, vm, currentNetName)
+	if err != nil {
+		return err
+	}
+
+	glog.V(6).Infof("changing network `%s` to `%s` for vm `%s`", currentBacking.DeviceName, newNetName, vm.Name())
+	currentBacking.DeviceName = newNetName
+	currentBacking.Network = newNet
+
+	err = vm.EditDevice(ctx, netDev)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getNetworkDeviceAndBackingFromVM(ctx context.Context, vm *object.VirtualMachine, netName string) (types.BaseVirtualDevice, *types.VirtualEthernetCardNetworkBackingInfo, error) {
+	devices, err := vm.Device(ctx)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't get devices for vm, see: %s", err)
+	}
+
+	var availableBackings []string
+
+	for _, device := range devices {
+		ethDevice, ok := device.(types.BaseVirtualEthernetCard)
+		if !ok {
+			continue
+		}
+
+		ethCard := ethDevice.GetVirtualEthernetCard()
+		ethBacking := ethCard.Backing.(*types.VirtualEthernetCardNetworkBackingInfo)
+
+		// if no network name is specified the first will be returned.
```

Comment:
I would be more comfortable if this func just returns a list of backings and devices in case `netName == ""`. If I am not mistaken this is also how the builtin funcs in `govmi` do it